lbrynet/dht/node.py

```diff
@@ -155,10 +149,17 @@ def __init__(self, node_id=None, udpPort=4000, dataStore=None,
         self.peer_finder = peer_finder or DHTPeerFinder(self, self.peer_manager)
         self._join_deferred = None
 
-    def __del__(self):
-        log.warning("unclean shutdown of the dht node")
-        if hasattr(self, "_listeningPort") and self._listeningPort is not None:
-            self._listeningPort.stopListening()
+    #def __del__(self):
+    #    log.warning("unclean shutdown of the dht node")
+    #    if hasattr(self, "_listeningPort") and self._listeningPort is not None:
+    #        self._listeningPort.stopListening()
+
+    def __str__(self):
+        return '<%s.%s object; ID: %s, IP address: %s, UDP port: %d>' % (
+            self.__module__, self.__class__.__name__, binascii.hexlify(self.node_id), self.externalIP, self.port)
+
+    def __hash__(self):
+        return int(binascii.hexlify(self.node_id), 16)
```

Comment:
why this, and why not the node id bytes?

Reply:
It has to return an `int`. That just helps converting it. `struct.unpack` doesn't support 384 bits afaik.